# **INDIAN STARTUP ECOSYSTEM**

## Project Description
We embark on a journey of discovery as we leverage our data analysis expertise to uncover the untapped potential within the Indian startup ecosystem. This project is designed to not only decode the numbers but to distill insights that will guide our team towards a successful foray into this dynamic market.

## Scope of Work

- Conduct a thorough exploration of datasets, dissecting funding patterns, sectoral nuances, and geographical hotspots in the Indian startup landscap
- Analyze funding received by startups in india from 2018 to 2021



## Hypothesis 

**Null Hypothesis (H0)**: There is no significant relationship between funding and the sector  

**Alternative Hypothesis (H1)**: There is a significant relationship between funding and the sector

## Questions 
1. How does funding vary across different industry sectors in India?
2. How does funding vary with the loaction of the start-ups
3. What is the relationship between the amount of funding and the stage of the company?
4. How have funding trends evolved between 2018 and 2021?
5. What are the most attractive sectors for investors?
6. Does the location of the company influence its sector?




# **DATA EXPLORATION, DATA UNDERSTANDING and DATA ANALYSIS**

In [1]:
# Load libraries
# Database connnection
import pyodbc     
from dotenv import dotenv_values    #import the dotenv_values function from the dotenv package

# Data manipulation
import pandas as pd
import numpy as np

# Data visualization
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

# filter warnings
import warnings 
warnings.filterwarnings('ignore')

# **1. Loading and Inspection of Data**

## **1.1 Loading data from the SQL server**

In [2]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER")
database = environment_variables.get("DATABASE")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

In [3]:
# Create a connection string

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"
    
connection = pyodbc.connect(connection_string)



In [4]:
# sql query to get 2020 data. 
query_2020="SELECT * FROM dbo.LP1_startup_funding2020"

# sql query to get 2021 data. 
query_2021="SELECT * FROM dbo.LP1_startup_funding2021"

In [5]:
    # load 2021 data
data_2021=pd.read_sql(query_2021,connection)

    # load 2020 data
data_2020=pd.read_sql(query_2020,connection)

## **1.2 Loading CSV Files**

In [6]:
# load 2019 data
data_2019=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2019.csv')

    # load 2018 data
data_2018=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2018.csv')


# **2.Exploratory Data Analaysis(EDA)**

## **2.0.  2018 Dataset EDA**

In [7]:
# preview the rows and columns for the 2018 dataset
data_2018.sample(5)

Company Name  \
372         Stelis Biopharma   
208                    Streo   
231            True Elements   
286  Sports Flashes Pvt. Ltd   
308                 Doubtnut   

                                              Industry Round/Series  \
372                                          Biopharma     Series B   
208                                    Music Streaming         Seed   
231  Dietary Supplements, Food and Beverage, Health...         Seed   
286  Basketball, Cricket, Cycling, eSports, Fitness...         Seed   
308  E-Learning, EdTech, Education, Image Recogniti...         Seed   

           Amount                     Location  \
372  $100,000,000  Bangalore, Karnataka, India   
208             —      Gurgaon, Haryana, India   
231   ₹50,000,000     Pune, Maharashtra, India   
286       1000000      New Delhi, Delhi, India   
308   ₹35,000,000      Gurgaon, Haryana, India   

                                         About Company  
372  Stelis Biopharma is a vertically integrated bi...  
208  Live stream music from nightclubs, bars & fest...  
231  True elements offers a range of high-quality s...  
286  Sports Flashes operates a sports content app t...  
308                 Gurgaon based E-learning platform.

In [8]:
# checking for number of columns and rows
print (data_2018.shape)
print(f"There are {data_2018.shape[0]} rows, and {data_2018.shape[1]} columns")

(526, 6)
There are 526 rows, and 6 columns


In [9]:
# checking info
data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [10]:
# Describing the data
data_2018.describe()

Company Name Industry Round/Series Amount  \
count               526      526          526    526   
unique              525      405           21    198   
top     TheCollegeFever        —         Seed      —   
freq                  2       30          280    148   

                           Location  \
count                           526   
unique                           50   
top     Bangalore, Karnataka, India   
freq                            102   

                                            About Company  
count                                                 526  
unique                                                524  
top     TheCollegeFever is a hub for fun, fiesta and f...  
freq                                                    2

**Findings**  
- TheCollegeFever company was the most common company  

- Seed series was the most preffered  

- most companies were based in Bangalore location  

In [11]:
# checking for duplicates
print("There are ",data_2018.duplicated().sum(),"duplicate(s)")

There are  1 duplicate(s)


In [12]:
# Checking for missing values
data_2018.isnull().sum()

Company Name     0
Industry         0
Round/Series     0
Amount           0
Location         0
About Company    0
dtype: int64

## **2.1. 2019 EDA**

In [13]:
# get a sample of 2019 dataset
data_2019.sample(5)

Company/Brand  Founded HeadQuarter      Sector  \
18       Witblox   2014.0         NaN      Edtech   
14   Open Secret      NaN         NaN   Food tech   
16        Byju's   2011.0         NaN      Edtech   
9       Lenskart   2010.0       Delhi  E-commerce   
72      CarDekho   2008.0      Jaipur  Automobile   

                                    What it does  \
18     Offers a range of robotics learning tools   
14      It produces and sells top quality snacks   
16              Provides online learning classes   
9                        It is a eyewear company   
72  Enables people to buy and sell new/old cars.   

                                        Founders  \
18                                     Amit Modi   
14                   Ahana Gautam, Udit Kejriwal   
16                               Byju Raveendran   
9   Peyush Bansal, Amit Chaudhary, Sumeet Kapahi   
72                                     Amit Jain   

                                             Investor     Amount($)     Stage  
18                              Mumbai Angels Network      $182,700       NaN  
14                                    Matrix Partners   Undisclosed       NaN  
16  South Africa’s Naspers Ventures, the CPP Inves...  $540,000,000       NaN  
9                                            SoftBank  $275,000,000  Series G  
72  Ping An Global Voyager Fund, Hillhouse Capital...   $70,000,000  Series D

In [14]:
# checking for number of columns and rows
print (data_2019.shape)
print(f"There are {data_2019.shape[0]} rows, and {data_2019.shape[1]} columns")

(89, 9)
There are 89 rows, and 9 columns


In [15]:
# checking for duplicates
print("There are ",data_2019.duplicated().sum(),"duplicate(s)")

There are  0 duplicate(s)


In [16]:
# Checking for nulls
data_2019.isnull().sum()

Company/Brand     0
Founded          29
HeadQuarter      19
Sector            5
What it does      0
Founders          3
Investor          0
Amount($)         0
Stage            46
dtype: int64

In [17]:
# checking for datatypes in the different columns
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [18]:
# performing descriptive analysis
data_2019.describe(include='all')

Company/Brand      Founded HeadQuarter  Sector      What it does  \
count             89    60.000000          70      84                89   
unique            87          NaN          17      52                88   
top         Kratikal          NaN   Bangalore  Edtech  Online meat shop   
freq               2          NaN          21       7                 2   
mean             NaN  2014.533333         NaN     NaN               NaN   
std              NaN     2.937003         NaN     NaN               NaN   
min              NaN  2004.000000         NaN     NaN               NaN   
25%              NaN  2013.000000         NaN     NaN               NaN   
50%              NaN  2015.000000         NaN     NaN               NaN   
75%              NaN  2016.250000         NaN     NaN               NaN   
max              NaN  2019.000000         NaN     NaN               NaN   

                          Founders     Investor    Amount($)     Stage  
count                           86           89           89        43  
unique                          85           86           50        15  
top     Vivek Gupta, Abhay Hanjura  Undisclosed  Undisclosed  Series A  
freq                             2            3           12        10  
mean                           NaN          NaN          NaN       NaN  
std                            NaN          NaN          NaN       NaN  
min                            NaN          NaN          NaN       NaN  
25%                            NaN          NaN          NaN       NaN  
50%                            NaN          NaN          NaN       NaN  
75%                            NaN          NaN          NaN       NaN  
max                            NaN          NaN          NaN       NaN

**Findings**
- Kratikal company was the most common company

- Most companies had their headquarters based in Bangalore

- Majority of the companies are involved in the Edtech sector

## **2.2. 2020 EDA**

In [19]:
# get a sample of 2020 dataset
data_2020.sample(5)

Company_Brand  Founded HeadQuarter      Sector  \
730  Biryani By Kilo      NaN       Delhi        Food   
710       Shiprocket   2017.0        None    Logitech   
920          Vedantu      NaN   Bangalore      Edtech   
853        Innerchef      NaN     Gurgaon    Foodtech   
363       ConnectedH   2016.0     Gurgaon  HealthCare   

                                          What_it_does  \
730                   Online paltform to order biryani   
710                  Tech enabled logistics aggregator   
920                            Onine learning platform   
853                             Cloud kitchen platform   
363  The company’s vision is simple – Connected Hea...   

                                              Founders  \
730                                      Vishal Jindal   
710                                        Saahil Goel   
920  Saurabh Saxena, Pulkit Jain, Anand Prakash, Va...   
853                                             Rajesh   
363                        Shubham Gupta, Suresh Singh   

                                              Investor      Amount     Stage  \
730  Devanshu Dolat Desai, Anand M Khatau, Debasish...    790000.0  Series B   
710                Tribe Capital, Innoven Capital, BII  13000000.0  Series C   
920                                        GGV Capital  24000000.0  Series C   
853                                          Mistletoe   1000000.0      None   
363                                               None         NaN      None   

    column10  
730     None  
710     None  
920     None  
853     None  
363     None

In [20]:
#checking for number of columns and rows
print (data_2020.shape)
print(f"There are {data_2020.shape[0]} rows, and {data_2020.shape[1]} columns")

(1055, 10)
There are 1055 rows, and 10 columns


In [21]:
# checking for duplicates
print("There are ",data_2020.duplicated().sum(),"duplicate(s)")

There are  3 duplicate(s)


In [22]:
# Checking for nulls
data_2020.isnull().sum()

Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [23]:
# checking for datatypes in the different columns
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [24]:
# Describe the data
data_2020.describe(include='all')

Company_Brand      Founded HeadQuarter   Sector  \
count           1055   842.000000         961     1042   
unique           905          NaN          77      302   
top            Nykaa          NaN   Bangalore  Fintech   
freq               6          NaN         317       80   
mean             NaN  2015.363420         NaN      NaN   
std              NaN     4.097909         NaN      NaN   
min              NaN  1973.000000         NaN      NaN   
25%              NaN  2014.000000         NaN      NaN   
50%              NaN  2016.000000         NaN      NaN   
75%              NaN  2018.000000         NaN      NaN   
max              NaN  2020.000000         NaN      NaN   

                            What_it_does       Founders           Investor  \
count                               1055           1043               1017   
unique                               990            927                848   
top     Provides online learning classes  Falguni Nayar  Venture Catalysts   
freq                                   4              6                 20   
mean                                 NaN            NaN                NaN   
std                                  NaN            NaN                NaN   
min                                  NaN            NaN                NaN   
25%                                  NaN            NaN                NaN   
50%                                  NaN            NaN                NaN   
75%                                  NaN            NaN                NaN   
max                                  NaN            NaN                NaN   

              Amount     Stage  column10  
count   8.010000e+02       591         2  
unique           NaN        42         2  
top              NaN  Series A  Pre-Seed  
freq             NaN        96         1  
mean    1.130430e+08       NaN       NaN  
std     2.476635e+09       NaN       NaN  
min     1.270000e+04       NaN       NaN  
25%     1.000000e+06       NaN       NaN  
50%     3.000000e+06       NaN       NaN  
75%     1.100000e+07       NaN       NaN  
max     7.000000e+10       NaN       NaN

**Findings**  
- Nykaa company was the most common company  

- Majority of the companies had their headquarter at Bangalore  

- The average Amount invested was 113042969.54

## **2.3. 2021 EDA**

In [25]:
# get a sample of 2021 dataset
data_2021.sample(5)

Company_Brand  Founded HeadQuarter        Sector  \
1110        CARS24   2015.0    Gurugram    Automotive   
268   Mohalla Tech   2015.0   Bangalore  Social media   
58            CRED   2018.0   Bangalore       FinTech   
857        6Degree   2014.0      Mumbai        Retail   
188       StockGro   2020.0   Bangalore       FinTech   

                                           What_it_does  \
1110  CARS24 is proud to be a tech-first organisatio...   
268   Mohalla Tech Pvt. Ltd. is a computer software ...   
58    CRED is a members-only credit card management ...   
857   6Degree is a fashion retail startup that provi...   
188   Making Investment Social- India's First Social...   

                                 Founders  \
1110         Mehul Agrawal, Vikram Chopra   
268   Ankush Sachdeva, Bhanu Pratap Singh   
58                             Kunal Shah   
857           Amit Bhardwaj, Nikhil Hegde   
188                         Ajay Lakhotia   

                                               Investor        Amount  \
1110    DST Global, Falcon Edge, SoftBank Vision Fund 2    $450000000   
268                                      Alkeon Capital    $266000000   
58    Dragoneer Investment Group, Tiger Global Manag...  $215,000,000   
857            Keiretsu Forum, SucSEED Venture Partners    $1,000,000   
188                       Roots Ventures, Velo Partners      $5000000   

             Stage  
1110      Series F  
268       Series G  
58        Series D  
857           None  
188   Pre-series A

In [26]:
# get the number of rows and columns for the datasets
print(f'{data_2021.shape}')

print(f"The 2021 dataset has {data_2021.shape[0]} rows and {data_2021.shape[1]} Columns")

(1209, 9)
The 2021 dataset has 1209 rows and 9 Columns


In [27]:
# checking for duplicates
print("There are ",data_2021.duplicated().sum(),"duplicate(s)")

There are  19 duplicate(s)


In [28]:
#  Checking for nulls
data_2021.isnull().sum()

Company_Brand      0
Founded            1
HeadQuarter        1
Sector             0
What_it_does       0
Founders           4
Investor          62
Amount             3
Stage            428
dtype: int64

In [29]:
# checking for datatypes in the different columns
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


In [30]:
# Describe the dataset
data_2021.describe(include='all')

Company_Brand      Founded HeadQuarter   Sector  \
count           1209  1208.000000        1208     1209   
unique          1033          NaN          70      254   
top         BharatPe          NaN   Bangalore  FinTech   
freq               8          NaN         426      122   
mean             NaN  2016.655629         NaN      NaN   
std              NaN     4.517364         NaN      NaN   
min              NaN  1963.000000         NaN      NaN   
25%              NaN  2015.000000         NaN      NaN   
50%              NaN  2018.000000         NaN      NaN   
75%              NaN  2020.000000         NaN      NaN   
max              NaN  2021.000000         NaN      NaN   

                                             What_it_does  \
count                                                1209   
unique                                               1143   
top     BharatPe develops a QR code-based payment app ...   
freq                                                    4   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

                                Founders                   Investor  \
count                               1205                       1147   
unique                              1095                        937   
top     Ashneer Grover, Shashvat Nakrani  Inflection Point Ventures   
freq                                   7                         24   
mean                                 NaN                        NaN   
std                                  NaN                        NaN   
min                                  NaN                        NaN   
25%                                  NaN                        NaN   
50%                                  NaN                        NaN   
75%                                  NaN                        NaN   
max                                  NaN                        NaN   

              Amount Stage  
count           1206   781  
unique           278    31  
top     $Undisclosed  Seed  
freq              73   246  
mean             NaN   NaN  
std              NaN   NaN  
min              NaN   NaN  
25%              NaN   NaN  
50%              NaN   NaN  
75%              NaN   NaN  
max              NaN   NaN

**Findings**  
- BharatPe is the most common company  

- Most companies invested in the FinTech sector  

- Majority of the companies were found in Bangalore  


**Overall Observations:**  
**Issues with the data**

1. There is a discrepancy in the naming conventions between the columns in the 2018 and 2019 datasets compared to the 2020 and 2021 datasets.

2. The 2018 dataset exhibits some missing columns, contributing to an incomplete representation of the data.

3. Conversely, the 2020 dataset contains an additional column that appears to be extraneous and does not serve a meaningful purpose in our analysis.

**Course of Action:**

1. **Missing Column Engineering for 2018:**
   - We will address the absence of certain columns in the 2018 dataset by employing data engineering techniques to create and populate the missing columns, ensuring a comprehensive and consistent dataset.

2. **Column Name Standardization:**
   - To establish uniformity and coherence across all datasets, we will embark on a column renaming process for the 2018 and 2019 datasets. This action aims to align the naming conventions with those observed in the 2020 and 2021 datasets, facilitating seamless data integration and analysis.

3. **Extraneous Column Removal in 2020:**
   - The redundant column identified in the 2020 dataset will be removed, streamlining the dataset and eliminating unnecessary elements that do not contribute to the overall analysis objectives.

These actions collectively enhance the integrity, consistency, and completeness of the dataset, paving the way for a more robust and coherent analytical process.
ical process.







# **3. Data Cleaning and Preparation**

## **3.1. Convert Indian Rupee values in the 2018 data to dollars**


In [31]:
# Replace '₹', commas, '—', and "''" in 'Amount' column
data_2018['Amount'] = data_2018['Amount'].str.replace(',', '').str.replace('—', '').str.replace("''",'').replace('', np.nan)

# Conditionally apply multiplication only where '₹' is present
mask = data_2018['Amount'].str.contains('₹', na=False)
data_2018.loc[mask, 'Amount'] = data_2018.loc[mask, 'Amount'].str.replace('₹', '').astype(float) * 0.0146

### **Add year data was collected**

- There is need to add a column that represents the year each dataset was collected. This will help with handling the datasets after merging the dataframes 


In [32]:
# add year when data was collected column to every dataset
data_2018['year'] = pd.to_datetime('2018', format='%Y').year
data_2019['year'] = pd.to_datetime('2019', format='%Y').year
data_2020['year'] = pd.to_datetime('2020', format='%Y').year
data_2021['year'] = pd.to_datetime('2021', format='%Y').year

In [33]:
def concat_dataframes(data,data1, data2, data3):
    """
    Concatenates four dataframes along the row axis.

    Parameters:
    data (pandas.DataFrame): The first dataframe to concatenate.
    data1 (pandas.DataFrame): The second dataframe to concatenate.
    data2 (pandas.DataFrame): The third dataframe to concatenate.
    data3 (pandas.DataFrame): The fourth dataframe to concatenate.

    Returns:
    pandas.DataFrame: The concatenated dataframe.
    """
    # Rename columns to make them consistent across dataframes
    df1 = data.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded','HeadQuarter': 'headquarter','Sector': 'sector','What_it_does': 'about_company','Founders':'founders','Investor':'investor','Amount':'amount','Stage':'series', 'year' : 'year'})
    df2 = data1.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded','HeadQuarter': 'headquarter','Sector': 'sector','What_it_does': 'about_company','Founders':'founders','Investor':'investor','Amount':'amount','Stage':'series','year' : 'year'})
    df3 = data2.rename(columns={'Company/Brand': 'company_brand', 'Founded': 'founded','HeadQuarter': 'headquarter','Sector': 'sector','What it does': 'about_company','Founders':'founders','Investor':'investor','Amount($)':'amount','Stage':'series','year' : 'year'})
    df4 = data3.rename(columns={'Company Name': 'company_brand','Location': 'headquarter','Industry': 'sector','About Company': 'about_company','Amount':'amount','Round/Series':'series','year' : 'year'})

    # Concatenate dataframes along the row axis
    result = pd.concat([df1, df2, df3,df4], axis=0).reset_index(drop=True)

    return result

## **Concatenate the dataframes**

In [34]:
# concating the dataframes
df= concat_dataframes(data_2021,data_2020,data_2019,data_2018)

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2879 entries, 0 to 2878
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_brand  2879 non-null   object 
 1   founded        2110 non-null   float64
 2   headquarter    2765 non-null   object 
 3   sector         2861 non-null   object 
 4   about_company  2879 non-null   object 
 5   founders       2334 non-null   object 
 6   investor       2253 non-null   object 
 7   amount         2474 non-null   object 
 8   series         1941 non-null   object 
 9   year           2879 non-null   int64  
 10  column10       2 non-null      object 
dtypes: float64(1), int64(1), object(9)
memory usage: 247.5+ KB


In [36]:
#Drop the extreneous column 10
df.drop('column10', axis=1, inplace= True)

## **Cleaning 'Amount' column**

**Notes**  
- Remove all currency signs  

- Remove all other umwanted characters, words and symbols  

- Fill the nulls uning interpolate method 

- Convert the column from object to float

In [37]:
df['amount']=df['amount'].replace('\$','',regex=True)
df['amount']=df['amount'].replace(['Undisclosed','undisclosed',None],np.NAN)
df['headquarter']=df['headquarter'].str.strip('\t#REF!')
df['amount']=df['amount'].replace('\,','',regex=True)
df['amount']=df['amount'].replace(",","")



In [47]:
# fix index 242
df.at[242, 'headquarter'] = None
df.at[242, 'sector'] = 'Pharmaceuticals'
df.at[242, 'founders'] = 'Varun Khanna'
df.at[242, 'investor'] = 'Morgan Stanley Private Equity Asia'
df.at[242, 'amount'] = '22000000'
df.at[242, 'series'] = 'Series C'
df.at[242, 'about_company'] = 'Development and Manufacturing'

# drop the duplicate row
df = df.drop(256, axis=0, errors='ignore')

# fix index 1148
df.at[1148, 'sector'] = 'Logistics'
df.at[1148, 'founders'] ='Basant Kumar' 
df.at[1148, 'investor'] =None
df.at[1148, 'amount'] = 1000000
df.at[1148, 'series'] = 'Seed'
 
# fix index 257
df.at[257, 'sector'] = 'Transport'
df.at[257, 'founders'] ='Anshuman Maheshwary'
df.at[257, 'investor'] =None
df.at[257, 'amount'] = 5000000
df.at[257, 'series'] = 'Seed'

# fix index 98
df.at[98, 'sector'] = 'Computer Games'
df.at[98, 'founders'] ='Pritesh Kumar'
df.at[98, 'investor'] =None
df.at[98, 'amount'] = 1200000
df.at[98, 'series'] = 'Seed'

# drop the duplicate column
df = df.drop(111, axis=0, errors='ignore')

# fix index 677
df.at[677, 'amount'] = 1000000
df.at[677, 'series'] = 'Angel'

# fix index 545
df.at[545, 'investor'] =None
df.at[545, 'amount'] = 1000000
df.at[545, 'series'] = 'Pre-series A'

# fix index 538
df.at[538, 'investor'] =None
df.at[538, 'amount'] = 300000
df.at[538, 'series'] = 'Pre-Seed'

# fix index 551
df.at[551,'amount']=300000
df.at[551,'series']=None

# fix index 674
df.at[674,'amount']=6000000
df.at[674,'series']=None

# fix index 2531
df.at[2531,'amount']=None
df.at[2531,'series']=None


In [50]:
# change the amount column to numeric
df['amount']=pd.to_numeric(df['amount'])

## **Cleaning data_year column**

**Notes**  


- Convert data type to period


In [51]:
# Convert the data_year column to date
df['year']=pd.to_datetime(df['year'], format='%Y')
df['year']=df['year'].dt.to_period('y')
df['founded']=pd.to_datetime(df['founded']).dt.year

In [52]:
# check for nulls and duplicated
print(f"There are {df['year'].isna().sum()} Null values in the 'year' column")

There are 0 Null values in the 'year' column


In [53]:
df['year'].info()

<class 'pandas.core.series.Series'>
Index: 2877 entries, 0 to 2878
Series name: year
Non-Null Count  Dtype        
--------------  -----        
2877 non-null   period[Y-DEC]
dtypes: period[Y-DEC](1)
memory usage: 109.5 KB


## **Cleaning 'founded' column**

**Notes**
- Handle nulls by populating with the 'bfill' method



In [54]:
print(f"There are {df['founded'].isna().sum()} Null values in the 'founded' column")

There are 769 Null values in the 'founded' column


**Notes**  
- There are 769 null values in the 'founded' column.  

- Since dropping the nulls will lead to a significant loss of our data, Backward fill will be used to fill the null values

**COURSE OF ACTION**

- We will be cleaning the missing values by fill them using interpolate method which is suitable for time series data  

- We will also be converting the data type from float to datetime for purpose of our analysis.

In [55]:
# Fill the nulls
df["founded"].interpolate(method='linear',inplace= True)

print(f"There are {df['founded'].isna().sum()} missing values")

There are 0 missing values


In [56]:
# Convert to datetime
df['founded'] = pd.to_datetime(df['founded'], format='%Y')

# Convert to period
df['founded'] = df['founded'].dt.to_period('Y')


In [57]:
df['founded'].info()

<class 'pandas.core.series.Series'>
Index: 2877 entries, 0 to 2878
Series name: founded
Non-Null Count  Dtype        
--------------  -----        
2877 non-null   period[Y-DEC]
dtypes: period[Y-DEC](1)
memory usage: 109.5 KB


In [58]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2877 entries, 0 to 2878
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype        
---  ------         --------------  -----        
 0   company_brand  2877 non-null   object       
 1   founded        2877 non-null   period[Y-DEC]
 2   headquarter    2762 non-null   object       
 3   sector         2859 non-null   object       
 4   about_company  2877 non-null   object       
 5   founders       2332 non-null   object       
 6   investor       2246 non-null   object       
 7   amount         2320 non-null   float64      
 8   series         1941 non-null   object       
 9   year           2877 non-null   period[Y-DEC]
dtypes: float64(1), object(7), period[Y-DEC](2)
memory usage: 311.8+ KB


## **Cleaning the 'founders' column**

In [59]:
# Remove unwanted characters
df['founders'] = df['founders'].replace(['...', np.nan], np.NaN)

# Check the number of NaN values in the 'founders' column
nan_count = df['founders'].isna().sum()

print(nan_count)

545


In [60]:
df['founders'].info()

<class 'pandas.core.series.Series'>
Index: 2877 entries, 0 to 2878
Series name: founders
Non-Null Count  Dtype 
--------------  ----- 
2332 non-null   object
dtypes: object(1)
memory usage: 109.5+ KB


## **Cleaning **Stage** column**

Startups start with pre-seed, progress through seed, Series A, Series B, etc., securing resources for development and strategies. Additional rounds like Series C or D may follow. External funding at each stage fuels growth toward the venture's full potential.

**Pre-Seed Funding**  
Entrepreneurial idea in early development; small funds needed; limited informal channels for raising funds.

**Seed Funding**  
First official equity funding; investors provide funds for equity ownership.

**Series A Financing**  
First venture capital round; developed product, consistent revenue, long-term profit plan.

**Series B Financing**  
For established startups; substantial user base and revenue; funding for expansion.

**Series C and Beyond**  
Optional rounds for final push before IPO or unmet objectives; Series C is the third venture capital round.

**Initial Public Offering (IPO)**  
Process of offering corporate shares to the public; used for funding or divestment.

link: https://www.startupindia.gov.in/content/sih/en/funding.html

In [61]:
# fix index 551
df.at[551,'amount']=300000
df.at[551,'series']=None

# fix index 674
df.at[674,'amount']=6000000
df.at[674,'series']=None

# fix index 2531
df.at[2531,'amount']=None
df.at[2531,'series']=None

In [62]:
# Standardize funding stages in the 'stage' column
df['series'] = df['series'].replace(['Series A', 'Seies A', 'Series A-1', 'Series A2', 'Series A+', 'Series A+'], 'Series A')
df['series'] = df['series'].replace(['Pre-seed', 'Pre-seed Round', 'Pre seed Round', 'Pre seed round'], 'Pre-Seed Stage')
df['series'] = df['series'].replace(['Pre series A', 'Pre-series A', 'Pre Series A', 'Pre series A1', 'Pre-series A1', 'Pre- series A'], 'Pre series A')
df['series'] = df['series'].replace(['Series B', 'Series B+', 'Series B2', 'Series B3'], 'Series B')
df['series'] = df['series'].replace(['Series C', 'Series C', 'Series C, D','Series C', 'Private Equity','PE', 'Post-IPO Equity','Series D', 'Series E', 'Series F', 'Series G', 'Series H', 'Series I','Series D1','Series F2', 'Series F1'], 'Series C and Beyond')
df['series'] = df['series'].replace(['Venture - Series Unknown', None,'Grant','Debt','Debt Financing','Post-IPO Debt','Non-equity Assistance','Bridge','Bridge Round','Fresh funding','Funding Round','Mid series','Edge',], 'unknown')
df['series'] = df['series'].replace(['Corporate Round','Undisclosed','Secondary Market','Pre-series','Post series A','Pre-series B','Pre-Series B','Pre series B','Pre-series C','Pre series C'], 'Other Stages')
df['series'] = df['series'].replace(['Seed','Seed funding','Pre-Seed','Angel', 'Angel Round','Seed fund', 'Seed round', 'Seed A','Seed Funding', 'Seed Round & Series A', 'Series E2', 'Seed Round','Seed Investment','Seed+','Early seed'],'Seed Stage')

In [63]:
# strip off characters 
df['series'] = df['series'].str.strip('\t#REF!')

## **Cleaning the Sector Column**

In [64]:
# Get the first sentence of every list
df['sector']=df['sector'].str.split(",").str[0]

In [65]:
df['sector'] = df['sector'].replace({
    'Edtech': 'EdTech', 'Fintech': 'FinTech', 'Agriculture': 'AgriTech', 'Food & Beverages': 'Food and Beverages',
    'Financial Services': 'FinTech', 'Healthcare': 'HealthTech', 'HealthTech': 'HealthTech', 'Medical': 'HealthTech', 
    'Medtech': 'HealthTech', 'Pharmaceutical': 'HealthTech', 'Health Insurance': 'HealthTech', 
    'Biotechnology': 'HealthTech', 'Health Diagnostics': 'HealthTech', 'Hospital': 'HealthTech', 
    'Hospital & Health Care': 'HealthTech', 'Wellness': 'HealthTech', 'Dental': 'HealthTech', 
    'Alternative Medicine': 'HealthTech', 'Nutrition': 'HealthTech', 'Fitness': 'HealthTech', 
    'Mental Health': 'HealthTech', 'Healthcare/Edtech': 'HealthTech',
    'Life sciences': 'HealthTech', 'Biotech': 'Healthcare', 'Nutrition Tech': 'HealthTech', 
    'E-mobility': 'HealthTech', 'Med Tech': 'HealthTech', 'FemTech': 'HealthTech', 
    'Cannabis startup': 'HealthTech', 'Pharmacy': 'HealthTech', 'Medical Device': 'HealthTech', 
    'BioTechnology': 'HealthTech', 'Fertility tech': 'HealthTech', 'Ayurveda tech': 'HealthTech', 
    'E-tail': 'Healthcare', 'E store': 'E-Commerce', 'E-store': 'Healthcare', 'Telemedicine': 'HealthTech', 
    'HealthCare': 'HealthTech', 'AI startup': 'AI', 'Information Services': 'InfoTech & Services', 
    'Healthtech': 'HealthTech', 'Finance': 'FinTech', 'Health Care': 'HealthTech', 
    'Logistics & Supply Chain': 'Logistics', 'Food Industry': 'FoodTech', 'Foodtech': 'FoodTech', 
    '—': 'Undisclosed', 'SaaS startup': 'SaaS', 'Health': 'HealthTech', 'Ecommerce': 'E-Commerce', 
    'Tech Startup': 'Tech', 'Mobility': 'Transportation', 'SaaS': 'Tech', 'Artificial Intelligence': 'AI', 
    'Food and Beverage': 'Food and Beverages', 'Information Technology': 'InfoTech & Services', 
    'Internet': 'Tech', 'Apps': 'Tech', 'Computer Software': 'Tech', 'E-commerce': 'E-Commerce', 
    'Agritech': 'AgriTech', 'Hospital & Health Care': 'HealthTech', 'Food': 'Foodtech', 'Cosmetics': 'Consumer Goods', 
    'Tech company': 'Tech', 'Automobile': 'Automotive', 'Apparel & Fashion': 'Fashion', 'Education': 'EdTech', 
    'Social Media': 'Media', 'Digital Media': 'Media', 'IT': 'InfoTech & Services', 'IoT': 'AI', 
    'Software': 'Tech', 'Industrial Automation': 'AI', 'Technology': 'Tech', 
    'Information Technology & Services': 'InfoTech & Services', None: 'Unknown'
})




In [ ]:
df[]

## **Cleaning HeadQuater Column**

In [66]:
# Get the first location from every list
df['headquarter']=df['headquarter'].str.split(",").str[0]

In [67]:
df['headquarter']=df['headquarter'].replace({'Bengaluru': 'Bangalore', 'Banglore': 'Bangalore', 'Gurugram': 'Gurgaon', 'Hyderebad': 'Hyderabad', 
                                      'New Delhi': 'Delhi', 'Ahmadabad': 'Ahmedabad', 'Ernakulam': 'Cochin', 'Telugana': 'Telangana',
                                      'Rajastan': 'Rajasthan', 'San Franciscao': 'San Francisco', 'Samsitpur': 'Samastipur', 'Santra': 'Samtra',
                                      'Rajsamand': 'Rajasthan', 'Kerala': 'Kochi','The Nilgiris': 'Nilgiris', 'Gurugram\t#REF!': 'Gurgaon', 
                                      'California': 'San Francisco', 'San Francisco Bay Area': 'San Francisco', 'Hyderebad': 'Hyderabad',
                                      'Online Media\t#REF!': 'Unknown','Pharmaceuticals\t#REF!': 'Unknown',
                                      'Information Technology & Services':'Unknown' ,'Small Towns': 'Unknown','Orissia': 'Odisha', 
                                      'Santra': 'Samtra', 'Vadodara': 'Vadodara', 'Food & Beverages': 'Unknown', 'Bangaldesh': 'Bangladesh',}) 


In [68]:
df.head()

company_brand founded headquarter          sector  \
0  Unbox Robotics    1970   Bangalore              AI   
1          upGrad    1970      Mumbai          EdTech   
2     Lead School    1970      Mumbai          EdTech   
3         Bizongo    1970      Mumbai  B2B E-commerce   
4        FypMoney    1970     Gurgaon         FinTech   

                                       about_company  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        investor       amount  \
0                    BEENEXT, Entrepreneur First    1200000.0   
1       Unilazer Ventures, IIFL Asset Management  120000000.0   
2               GSV Ventures, Westbridge Capital   30000000.0   
3                         CDC Group, IDG Capital   51000000.0   
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    2000000.0   

                series  year  
0         Pre series A  2021  
1              unknown  2021  
2  Series C and Beyond  2021  
3  Series C and Beyond  2021  
4           Seed Stage  2021

In [83]:
# check for duplicates
df[df.duplicated()]


company_brand founded      headquarter                 sector  \
107            Curefoods    1970        Bangalore     Food and Beverages   
109             Bewakoof    1970           Mumbai                Fashion   
117       Advantage Club    1970           Mumbai                 HRTech   
119               Ruptok    1970            Delhi                FinTech   
243             Trinkerr    1970        Bangalore        Capital Markets   
244                Zorro    1970          Gurgaon         Social network   
245        Ultraviolette    1970        Bangalore             Automotive   
246           NephroPlus    1970        Hyderabad             HealthTech   
247              Unremot    1970        Bangalore    InfoTech & Services   
248          FanAnywhere    1970        Bangalore                FinTech   
249           PingoLearn    1970             Pune             E-learning   
250                 Spry    1970           Mumbai                  Music   
251              Enmovil    1970        Hyderabad    InfoTech & Services   
252        ASQI Advisors    1970           Mumbai                FinTech   
253   Insurance Samadhan    1970            Delhi              Insurance   
254      Evenflow Brands    1970           Mumbai         Consumer Goods   
255           MasterChow    1970  ood & Beverages              Hauz Khas   
1354           Krimanshi    1970          Jodhpur  Biotechnology company   
1414               Nykaa    1970           Mumbai         Consumer Goods   
1571              Byju’s    1970        Bangalore                 EdTech   
2701     TheCollegeFever    1970        Bangalore        Brand Marketing   

                                          about_company  \
107   Healthy & nutritious foods and cold pressed ju...   
109   Bewakoof is a lifestyle fashion brand that mak...   
117   Advantage Club is India's largest employee eng...   
119   Ruptok fintech Pvt. Ltd. is an online gold loa...   
243   Trinkerr is India's first social trading platf...   
244                Pseudonymous social network platform   
245   Create and Inspire the future of sustainable u...   
246   A vision and passion of redefining healthcare ...   
247       Unremot is a personal office for consultants!   
248                              Celebrity NFT platform   
249   PingoLearn offers language learning courses wi...   
250   THE BEST ROYALTY-FREE MUSIC TO CREATE TRENDING...   
251   Enmovil delivers a Business Intelligence Platf...   
252   Bringing Blockchain technology intro mainstrea...   
253   The insurance industry will help in representa...   
254   Identify and partner with the upcoming eCommer...   
255                 A ready-to-cook Asian cuisine brand   
1354  Krimanshi aims to increase rural income by imp...   
1414  Nykaa is an online marketplace for different b...   
1571  An Indian educational technology and online tu...   
2701  TheCollegeFever is a hub for fun, fiesta and f...   

                                           founders  \
107                                    Ankit Nagori   
109                                Prabhkiran Singh   
117              Sourabh Deorah, Smiti Bhatt Deorah   
119                                     Ankur Gupta   
243                 Manvendra Singh, Gaurav Agarwal   
244   Jasveer Singh, Abhishek Asthana, Deepak Kumar   
245             Subramaniam Narayan, Niraj Rajmohan   
246                                  Vikram Vuppala   
247                             Shiju Radhakrishnan   
248                     Varun Chaudhary, Amit Kumar   
249              Mohit Menghani, Shubham Maheshwari   
250                 Gaurav Dagaonkar, Meghna Mittal   
251      Ravi Bulusu, Nanda Kishore, Venkat Moganty   
252                                   Swapnil Pawar   
253                      Deepak Bhuvenshwari Uniyal   
254                   Utsav Agarwal, Pulkit Chhabra   
255                   Vidur Kataria, Sidhanth Madan   
1354                                   Nikhil Boh

In [85]:
# drop duplicates
df.drop_duplicates(keep='first', inplace=True)


In [86]:
# check for duplicates
print(f" There are {df.duplicated().sum()} duplicates")

 There are 0 duplicates


In [71]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2877 entries, 0 to 2878
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype        
---  ------         --------------  -----        
 0   company_brand  2877 non-null   object       
 1   founded        2877 non-null   period[Y-DEC]
 2   headquarter    2762 non-null   object       
 3   sector         2877 non-null   object       
 4   about_company  2877 non-null   object       
 5   founders       2332 non-null   object       
 6   investor       2246 non-null   object       
 7   amount         2320 non-null   float64      
 8   series         2877 non-null   object       
 9   year           2877 non-null   period[Y-DEC]
dtypes: float64(1), object(7), period[Y-DEC](2)
memory usage: 311.8+ KB
